main/src/cgeo/geocaching/ui/dialog/TimeDialog.java

```diff
@@ -45,6 +46,7 @@ public View onCreateView(final LayoutInflater inflater, final ViewGroup containe
         picker.setCurrentHour(date.get(Calendar.HOUR_OF_DAY));
         picker.setCurrentMinute(date.get(Calendar.MINUTE));
         picker.setOnTimeChangedListener(new TimePickerListener());
+        picker.setIs24HourView(Settings.get24HourStyle());
```

Comment:
No need for this extra method, you can use:

```java
   picker.setIs24HourView(DateFormat.is24HourFormat(getActivity());
```

to refer to the context of the enclosing activity.